pkg/binding/backend/interactive.go

```diff
@@ -98,6 +102,38 @@ loop:
 	return selectedName, selectedGVK, nil
 }
 
+// SelectNamespaceForServiceInstances returns the namespace which services instances should be listed from.
+// An empty return value means that service instances will be listed from the current namespace.
+// Otherwise, they will be listed from the specified namespace.
+func (o *InteractiveBackend) SelectNamespaceForServiceInstances(_ map[string]string, nsListProvider func() ([]string, error)) (string, error) {
+	option, err := o.askerClient.SelectNamespaceListOption()
+	if err != nil {
+		return "", err
+	}
+
+	switch option {
+	case asker.CurrentNamespace:
+		return "", nil
+	case asker.AllAccessibleNamespaces:
+		klog.V(2).Infof("Listing all projects/namespaces...")
+		var nsList []string
+		nsList, err = nsListProvider()
```

Comment:
IMO, the logic to list namespaces should be in the business layer because it's not something specific to the CLI layer. It's the stuff we want to get from the k8s cluster. Such handler function coming from CLI layer should be used when we want to execute some CLI specific logic, which is not the case here. WDYT?

Reply:
I agree with Dharmit here. Also, if we only need to list namespaces for the interactive mode, why not just do it in `InteractiveBackend.SelectNamespaceForServiceInstances`?

Reply:
I think I did it this way mainly because of the warning message I wanted to display in the CLI if odo could not list all the namespaces/projects (e.g., due to RBAC issues in Kubernetes clusters). But you're right - I'll handle this differently.

Reply:
> I think I did it this way mainly because of the warning message I wanted to display in the CLI if odo could not list all the namespaces/projects (e.g., due to RBAC issues in Kubernetes clusters). But you're right - I'll handle this differently.

Ah that makes sense, but we have been using log/status on business layer for quite a few packages, adding one more to the list won't make much difference :sweat_smile: